crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -317,7 +359,7 @@ fn create_txn_env(block_env: &BlockEnv, request: CallRequest) -> EthResult<TxEnv
     let CallFees { max_priority_fee_per_gas, gas_price } =
         CallFees::ensure_fees(gas_price, max_fee_per_gas, max_priority_fee_per_gas)?;
 
-    let gas_limit = gas.unwrap_or(block_env.gas_limit);
+    let gas_limit = gas.unwrap_or(block_env.gas_limit.min(U256::from(u64::MAX)));
```

Comment:
why is this needed?
this seems redundant because the block env comes preconfigured?

Reply:
https://github.com/paradigmxyz/reth/blob/a1d0b537a7549717956cb750d10045f353a9f8e6/crates/rpc/rpc/src/eth/cache.rs#LL225C49-L225C49

In this line, the cache initializes the `BlockEnv` to the default value, which uses `U256::MAX` for the gas limit. This results in an error when using the `NoopProvider` as `EvmEnvProvider`, as it doesn't modify this limit.